### DISCUSSION TITLE
<SessionProvider/> doesn't respond to session updates.

### DISCUSSION THREAD

**Discussion by ahmedjk34** [nextauthjs/next-auth#11977] [Help]
This is currently the code:

//layout.tsx
![image](https://github.com/user-attachments/assets/ff176256-835c-4476-86d1-4434370091cf)

I used to pass the session to the nav and UserModal as a prop, and it worked properly _(since props respond to direct changes in the parent)_, but then i started using `useSession()` hook, and with that, when I login into the app, and the session does update _(I can see that by logging the session in the route)_ the session inside the nav and the modal doesn't.
What are the possible solutions for this?

**Comment by k-taro56** [comment]
`layout.tsx` is not re-rendered

https://nextjs.org/docs/app/api-reference/file-conventions/layout

**Reply by ahmedjk34** [reply]
However, I added a console.log to `layout.tsx`, and when i login it does log the session with the user. 
As in on initial page render it gives `null`
And after i login and redirect it does log the session.

But still it doesn't get passed updated until i refresh (the session provider takes the old value)

BUT, if i pass the session as a prop, the updates session will be registered by the children

**Reply by ahmedjk34** [reply]
So why is that? And what are the solutions to keep using the <SessionProvider/> while having the session update for the children of the layout
Thank you for your time!
